R/board_ms365.R

```diff
@@ -0,0 +1,188 @@
+#' Use a OneDrive or Sharepoint document library as a board
+#'
+#' Pin data to a folder in Onedrive or a SharePoint Online document library
+#' using the Microsoft365R package.
+#'
+#' @inheritParams new_board
+#' @param drive A OneDrive or SharePoint document library object, as obtained
+#'   from Microsoft365.
+#' @param path Path to directory to store pins. Will be created if it
+#'   doesn't already exist.
+#' @param delete_by_item Whether to handle folder deletions on an item-by-item
+#'   basis, rather than deleting the entire folder at once. You may need to set
+#'   this to `TRUE` for a board in SharePoint Online or OneDrive for Business,
+#'   due to document protection policies that prohibit deleting non-empty
+#'   folders.
+#' @export
+#' @examples
+#' \dontrun{
+#' # A board in your personal OneDrive
+#' od <- Microsoft365R::get_personal_onedrive()
+#' board <- board_onedrive(od, "myboard")
+#' board %>% pin_write(iris)
+#'
+#' # A board in OneDrive for Business
+#' odb <- Microsoft365R::get_business_onedrive(tenant = "mytenant")
+#' board <- board_onedrive(odb, "myproject/board")
+#'
+#' # A board in a SharePoint Online document library
+#' sp <- Microsoft365R::get_sharepoint_site("my site", tenant = "mytenant")
+#' doclib <- sp$get_drive()
+#' board <- board_sharepoint(doclib, "general/project1/board")
+#' }
+board_ms365 <- function(drive, path, versioned = TRUE, cache = NULL, delete_by_item = FALSE) {
+  check_installed("Microsoft365R")
+
+  if (!inherits(drive, "ms_drive")) {
+    abort("`drive` must be a OneDrive or SharePoint document library object")
+  }
+
+  # try to create the board folder: ignore error if folder already exists
+  try(drive$create_folder(path), silent = TRUE)
+  folder <- drive$get_item(path)
+  if (!folder$is_folder()) {
+    abort("Invalid path specified")
+  }
+
+  cache <- cache %||% board_cache_path(paste0("ms365-", hash(folder$properties$id)))
+  new_board_v1("pins_board_ms365",
+    folder = folder,
+    cache = cache,
+    versioned = versioned,
+    delete_by_item = delete_by_item
+  )
+}
+
+#' @export
+#' @rdname board_ms365
+board_sharepoint <- board_ms365
```

Comment:
I'm a bit conflicted about these: on the one hand they are more explicit names, on the other hand, they do exactly the same thing which might be confusing for users. I think on the whole I'd prefer to stick with just `board_ms365`.